### PR TITLE
Add a new `pkg/sleep` package to sleep safely with cancelation

### PR DESCRIPTION
**Pull request by howardjohn**
**Please provide a description of this PR:**

### CODE REVIEW

File: pkg/sleep/sleep.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+func Until(ch <-chan struct{}, d time.Duration) bool {
+	timer := time.NewTimer(d)
+	select {
+	case <-ch:
+		if !timer.Stop() {
+			<-timer.C
+		}
+		return false
+	case <-timer.C:
+		return true
```
Comment (hzxuzhonghu) [inline pkg/sleep/sleep.go:39]: You need to stop the timer before return

File: pkg/sleep/sleep.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+package sleep
+
+import (
+	"context"
+	"time"
+)
+
+// Context sleeps for the given duration, or until the context is complete.
+// Returns true if the sleep completes the full duration
+func Context(ctx context.Context, d time.Duration) bool {
```
Comment (hzxuzhonghu) [inline pkg/sleep/sleep.go:24]: This name is confusing, indicates it returns a context.

File: cni/pkg/install/install.go [hunk 3]
```diff
@@ -297,7 +298,7 @@ func (in *Installer) watchSAToken(ctx context.Context, fileModified chan bool, e
 				if curToken != token {
 					fileModified <- true
 				}
-				time.Sleep(1 * time.Minute)
+				sleep.Context(ctx, 1 * time.Minute)
```
Comment (hzxuzhonghu) [inline cni/pkg/install/install.go:301]: Do we still need L291?

File: pkg/sleep/sleep.go (hunk 2 above)
Comment (howardjohn) [reply pkg/sleep/sleep.go:24]: Any suggestions? When you call it's `sleep.Context()` so it was pretty clear to me.

File: cni/pkg/install/install.go (hunk 3 above)
Comment (howardjohn) [reply cni/pkg/install/install.go:301]: it protects the first call I guess

File: pkg/sleep/sleep.go (hunk 2 above)
Comment (hzxuzhonghu) [reply pkg/sleep/sleep.go:24]: How about removing this function. ctx.Done() is simple to call

Comment (ramaraochavali) [reply pkg/sleep/sleep.go:24]: Either remove or name `UntilContext` ?

Comment (ramaraochavali) [reply pkg/sleep/sleep.go:24]: UntilContext may be better?

**Review by hzxuzhonghu — APPROVED** [review]

**Review by ericvn — APPROVED** [review]

**Pull request merged**